### DISCUSSION TITLE
게시글 개수 제한에 대한 질문

### DISCUSSION THREAD

**Discussion by siyeons** [morethanmin/morethan-log#93] [Q&A] [answered]
안녕하세요~ 이 라이브러리가 너무 좋아서 열심히 써보려고 하고있는데요.. 

`
 {filteredPosts.slice(0, 20).map((post) => (
          <PostCard key={post.id} post={post} />
))}
`

PostList의 이 부분으로 인해 20개 이상의 게시글은 보여지지 않는 것 같아요. 물론 필터링해서 보면 되긴 하지만.. 🥹
저는 임시로 일단 저 slice 구문을 제거하고 react-window를 적용해두었는데요! 
현재의 UI에선 가상스크롤보다는 무한스크롤이 더 적합해보이긴 하네요 ㅎㅅㅎ (실제 적용해본 UX를 기반으로 하면)
혹시 이 부분은 의도하신 부분일까요? 아니면 추가 개발을 예정에 두고 계신지.. 궁금합니다! 아니면 제가 무언가 잘못 쓰고있는지 알려주실 수 있을까요?
데모는 게시글이 20개가 넘지 않아 아직 문제가 없는 것 같긴합니다! 

그리고 추가적으로.. 속성값(ex. 태그)이 빠지면 런타임에러가 나더라구요..
pages/index.tsx부분에 에러 예외처리를
`
    return {
      props: {
        tags: {
          All: 0,
          tags: [],
        },
        posts: [],
      },
      revalidate: 1,
    }
`
조금 더 추가해두었더니 런타임 에러는 막을 수 있었습니다. 물론 빌드할 때 serverless function들이 제대로 동작안해서 배포는 안되지만요 😅 
스타일 정도는 확인할 수 있습니다.. ㅎㅎ 

좋은 라이브러리 만들어주셔서 정말 감사합니다~~!

**Comment by morethanmin** [comment, marked as answer]
안녕하세요! 몰댄로그에 관심 가져주시고 도움주셔서 감사합니다!

일단 `slice`부분은 제가 페이지네이션 처리하다가 같이 올라간 부분인 것 같습니다. 🥲 말씀 주신대로 이부분은 제거 해주는게 맞을 것 같아요. `react-window` 라는 라이브러리도 찾아보니 매우 좋은 것 같아요! 이부분은 한번 살펴보고 적용할 수 있으면 적용해보겠습니다. ㅎㅎ

태그 부분은 노션 데이터베이스에서 태그 속성을 제거하신 케이스를 말씀해주시는거라면 아마 에러가 발생하는게 맞을 듯 한데, 혹시 태그를 제거하고 싶으셨던 이유가 궁금합니다! 어떤 의도이신지 파악이되면 이부분도 morethan-log에서 처리해드릴 수 있는 부분일지, 아니면 어떠한 방식으로 처리할 수있을지 좀더 고민해볼 수 있을 것 같아요.

감사합니다 😄

**Reply by siyeons** [reply]
답변 감사합니다 🙇‍♀️!!
가상스크롤이 나을지 무한스크롤이 나을지는 잘 모르겠습니다 🥲 
저는 가상스크롤로 해서 임의로 스크롤바를 숨겨버렸거든요..(css로) 뭔가 부모의 높이를 따라가고싶은데 컴포넌트 내부적으로 스크롤을 또 가지니 ux가 어색한것같긴합니다 ㅜㅜ 

아!! 의도된거였군요! 
제가 태그를 하나 빼먹고 올린 게시글이 있었는데 전체 런타임에러가 나고 어떤 에러인지 정확히 파악을 못했던 적이 있어서 예외처리를 추가했었는데요, 
생각해보니 에러가 나도록 하는게 더 빠르게 파악할 수 있겠네요! 감사합니다!
